### PR TITLE
TableWorkspaceDisplay column sort changes the workspace in the ADS

### LINKED ISSUE #24547: TableWorkspaceDisplay should use algorithm SortTableWorkspace to reflect changes in the ADS
### Expected behavior
Sorting a column in a TableWorkspaceDisplay sorts the column in the underlying workspace as well.

### Actual behavior
Column is only sorted visually. Changes are not reflected. 

This change should also lead to a code reduction, as the comparison code from `WorkbenchTableWidgetItem` may be removed. However, check if this breaks sorting V3D's again, and if so fix the algorithm to successfully sort V3D.

### PR DESCRIPTION
**Pull request by dtasev**
**Description of work.**
Changes the table sorting to use SortTable/PeaksWorkspace, so that the sorting is reflected in the ADS.

**To test:**
Open table workspace, right click a column and sort. Open the workspace anew, and it should look the same as the old window - meaning it has been changed 

This PR is dependent on ~~#24563~~ 

Fixes #24547 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by martyngigg — CHANGES_REQUESTED** [review]
It looks to me as if Ascending and Descending are the wrong way around? Try sorting any column and it ends up the opposite way around to what you selected.

It's also not clear that no sorting occurs if I don't fully left click column first but this can be dealt with separately.

**Comment by DTasev** [comment]
> It's also not clear that no sorting occurs if I don't fully left click column first but this can be dealt with separately.

Issue to add status bars for better user messages: #24630

File: qt/python/mantidqt/widgets/tableworkspacedisplay/presenter.py [hunk 1]
```diff
@@ -157,11 +156,7 @@ def action_delete_row(self):
         selected_rows_list = [index.row() for index in selected_rows]
         selected_rows_str = ",".join([str(row) for row in selected_rows_list])
 
-        DeleteTableRows(self.model.ws, selected_rows_str)
-        # Reverse the list so that we delete in order from bottom -> top
-        # this prevents the row index from shifting up when deleting rows above
-        for row in reversed(selected_rows_list):
```
Comment (DTasev) [inline qt/python/mantidqt/widgets/tableworkspacedisplay/presenter.py]: It is no longer needed to remove the row from the view, as the DeleteTableRows alg triggers a replaceWorkspace ADS event

**Review by martyngigg — APPROVED** [review]
This looks good now. The model is taking care of the sorting and the underlying workspace is affected as expected.

**Pull request merged**